source/base/Allocator.ooc

```diff
@@ -0,0 +1,25 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+import Owner
+
+AbstractAllocator: abstract class {
```

Comment:
Thanks @shamanas
`use template classes `
:arrow_up: that was the first thing I though of, but I gave up after fighting some compiler errors :/ This here is not perfect, but suits my needs for now. You have some nice ideas here, I think we may revisit this someday.